src/bin/rdcore/kargs.rs

```diff
@@ -26,10 +28,14 @@ pub fn kargs(config: &KargsConfig) -> Result<()> {
         // the unwrap() here is safe because we checked in cmdline that one of them must be provided
         let mount =
             get_boot_mount_from_cmdline_args(&config.boot_mount, &config.boot_device)?.unwrap();
-        visit_bls_entry_options(mount.mountpoint(), |orig_options: &str| {
+        let changed = visit_bls_entry_options(mount.mountpoint(), |orig_options: &str| {
             modify_and_print(config, orig_options)
         })
         .context("visiting BLS options")?;
+
+        if changed && cfg!(target_arch = "s390x") {
+            runcmd!("zipl", "--target", mount.mountpoint())?;
```

Comment:
Not 100% sure if that's all we need. /cc @nikita-dubrovskii

Basing this off of `ignition-firstboot-complete`, which just calls `zipl`, except that here the boot partition isn't necessarily mounted on `/boot` so we need to point `zipl` at it: https://github.com/coreos/ignition/blob/3885b9f07c96baf7dcc12c85b914ded151c8fd6e/systemd/ignition-firstboot-complete.service#L26

Reply:
Also, if a bare `zipl` works, I'm confused by all the explicit arguments in `install_bootloader()`.  If zipl indeed knows how to read BLS configs, could it do that in the `install_bootloader()` case too?

Reply:
I'll check it.

Reply:
So, `--target` option doesn't work without specifying `--image` and/or others. what could be used is `--blsdir` , but in such case `/boot` shoud be `rw` accessible.

Reply:
> what could be used is `--blsdir` , but in such case `/boot` shoud be `rw` accessible. 

Huh interesting, does it actually need write access to `/boot`, or just for the boot partition to be mounted at `/boot`? If the former, what does it write there?